### DISCUSSION TITLE
New command `rome migrate` or `rome config migrate`

### DISCUSSION THREAD

**Discussion by ematipico** [rome/tools#1366] [Suggestions]
Now that Rome's configuration is growing and that are allowing multiple formats soon (YAML and TOML), I was thinking that should automatically supports the following cases:

## Config extension migration

```bash
rome config migrate-extension
```

This would allow the users to change their configuration file from one extension to another (e.g. from RJSON to JSON and vice versa).

## Config options migration

```bash
rome config migrate
```

This one is maybe to most important one. As we aim to do fast releases and change rapidly, this command would help the users to install a new version of Rome and port their configuration to the new format. For example, if one day we decided to change the name of a key, the `migrate` command could automatically do it for the user. Thanks to the fact we have our own parser, we can surgically update the config to the new version.

Plus, it should also maintain multiple versions and how a config file would change across versions. Having a command like this would help us maintainers to make more breaking changes to how we maintain the config file because we just need to run a simple command to fix the new version

**Comment by sebmck** [comment]
This would be awesome! `rome config migrate` would be well accompanied with a `rome upgrade` command. It could pull the latest version of Rome, then `rome config migrate` to convert the config.

**Reply by ematipico** [reply]
Sounds a good plan! I am going to start putting some foundation behind! Please share some thoughts if there are things that we should take into account!
